R/repair-column-names.R

```diff
@@ -0,0 +1,19 @@
+#' Check validity/format of a data.frame or tbl.
+#'
+#' Ensure the tbl or data.frame has legitimate and optionally unique
+#' column names.
+#' @param x tbl, tbl__df, or data.frame
+#' @param unique logical (default \code{TRUE}), force column names to
+#'     be unique (\code{TRUE}) or allow duplicates (\code{FALSE});
+#'     ignored if \code{check.names} is \code{FALSE}
+#' @return data.frame, with optionally different column names
+#' @export
+repair_names <- function(x, unique=TRUE) {
+    col_names <- colnames(x)
+    if (is.null(col_names)) {
+        col_names <- rep('', ncol(x))
```

Comment:
Unfortunately, make.names() doesn't allow tweaking its behavior. But good news is that we can use make.unique(sep = ""), which does most of the heavy lifting.

Not sorted column names in corner cases are fine, this is what make.unique() rightly does:

```
> make.unique(c("a", "b", "a", "a1"), sep = "")
[1] "a"  "b"  "a2" "a1"
```